accounts/static/accounts/js/profile.js

```diff
@@ -0,0 +1,392 @@
+  };
+
+  // Enable modal, buttons, and blur
+  friend_request_modal.style.display = "block";
+  accept_friend_button.style.display = "inline-block";
+  reject_friend_button.style.display = "inline-block";
+  overlay.classList.remove("hidden");
+}
+
+/**
+ * This function closes the modal for accepting or rejecting a friend request.
+ */
+function closeFriendRequestModal() {
+  overlay.classList.add("hidden");
+  friend_request_modal.style.display = "none";
+}
+
+/**
+ * This function sends a POST request to accept a received friend request.
+ * @param {String} friend_username the username of the user that sent the friend request
+ */
+async function confirmFriend(friend_username) {
+  // Make a POST request to add friend to friends list
+  const csrfToken = await getCSRFToken();
+  fetch("/accounts/accept/friend/", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "X-CSRFToken": csrfToken,
+    },
+    body: JSON.stringify({ username: friend_username }),
+  })
+    .then((response) => {
+      // Update response message styling
+      if (response.status == 200) {
+        accept_friend_button.style.display = "none";
+        reject_friend_button.style.display = "none";
+        friend_request_response.style.color = "green";
+      } else {
+        friend_request_response.style.color = "red";
+      }
+
+      return response.text();
+    })
+    .then((data) => {
+      // Display response message
+      friend_request_response.innerHTML = data;
+    })
+    .catch((error) => {
+      console.log(
+        "An error occured during the accept friend request: " + error
+      );
+    });
+}
+
+/**
+ * This function sends a POST request to reject a received friend request.
+ * @param {String} friend_username the username of the user that sent the friend request
+ */
+async function rejectFriend(friend_username) {
+  // Make a POST request to reject friend request
+  const csrfToken = await getCSRFToken();
+  fetch("/accounts/reject/friend/", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "X-CSRFToken": csrfToken,
+    },
+    body: JSON.stringify({
+      username: friend_username,
+      username_is_sender: true,
+    }),
+  })
+    .then((response) => {
+      // Update response message styling
+      if (response.status == 200) {
+        accept_friend_button.style.display = "none";
+        reject_friend_button.style.display = "none";
+        friend_request_response.style.color = "green";
+      } else {
+        friend_request_response.style.color = "red";
+      }
+
+      return response.text();
+    })
+    .then((data) => {
+      // Display response message
+      friend_request_response.innerHTML = data;
+    })
+    .catch((error) => {
+      console.log(
+        "An error occured during the reject friend request: " + error
+      );
+    });
+}
+
+/****************************************************
+ *
+ * Send friend request functionality
```

Comment:
this is delete functionality
also i thought we combined deleting and rejecting into one thing?

Reply:
Yeah the delete friend request and reject friend request got combined into the: /accounts/reject/friend/ path, the path name should probably be updated now that you mention it, but I kept the functions separate

Reply:
Oh shoot I see what happened now, I just had the path redirect to the delete_friend_request view, will fix that